src/database.ts

```diff
@@ -1307,6 +1394,58 @@ class Database extends GrpcServiceObject {
       callback!(err, transaction);
     });
   }
+
+  /**
+   * Query object for listing database operations.
+   *
+   * @typedef {object} GetDatabaseOperationsRequest
+   * @property {boolean} [autoPaginate=true] Have pagination handled
+   *     automatically.
+   * @property {number} [maxApiCalls] Maximum number of API calls to make.
+   * @property {number} [maxResults] Maximum number of items to return.
+   * @property {number} [pageSize] Maximum number of results per page.
+   * @property {string} [pageToken] A previously-returned page token
+   *     representing part of the larger set of results to view.
+   */
+  /**
+   * @typedef {array} GetDatabaseOperationsResponse
+   * @property {IOperation[]} 0 Array of {@link IOperation} instances.
+   * @property {object} 1 The full API response.
+   */
+  /**
+   * List pending and completed operations for the database.
+   *
+   * @see {@link Instance.getDatabaseOperations}
+   *
+   * @param query query object for listing database operations.
+   * @returns {Promise<GetDatabaseOperationsResponse>} when resolved, contains
+   *     a paged list of database operations.
+   *
+   * @example
+   * const {Spanner} = require('@google-cloud/spanner');
+   * const spanner = new Spanner();
+   * const instance = spanner.instance('my-instance');
+   * const database = instance.database('my-database');
+   * const [operations] = await database.getOperations();
+   */
+  async getOperations(
+    query?: GetDatabaseOperationsRequest
+  ): Promise<GetDatabaseOperationsResponse> {
+    // Create a query that lists database operations only on this database from
+    // the instance. Operation name will be prefixed with the database path for
+    // all operations on this database
+    let dbSpecificFilter = `name:${this.formattedName_}`;
+    if (query && query.filter) {
+      dbSpecificFilter = `(${dbSpecificFilter}) AND (${query.filter})`;
+    }
+    const dbSpecificQuery: GetDatabaseOperationsRequest = {
+      ...query,
+      filter: dbSpecificFilter,
+    };
+
+    return this.instance.getDatabaseOperations(dbSpecificQuery);
```

Comment:
@mayurkale22 or @olavloite if either of you could help me with this I'd really appreciate it. The promise returned here never resolves and I don't really understand why. The issue shows up with these [system test failures](https://source.cloud.google.com/results/invocations/c7563f47-e93a-472f-b7d9-7069fef2d5c2/targets/cloud-devrel%2Fclient-libraries%2Fnodejs%2Fpresubmit%2Fgoogleapis%2Fnodejs-spanner%2Fnode10%2Fsystem-test/log).

Reply:
@skuruppu I'll have a look.

Reply:
Nice find @olavloite :) Thank you so much for the fix.